components/constellation/timer_scheduler.rs

```diff
@@ -127,95 +75,55 @@ impl TimerScheduler {
         chan
     }
 
-    fn run_event_loop(&self) {
-        while let Some(thread) = self.receive_next_task() {
-            match thread {
-                Task::HandleRequest(request) => self.handle_request(request),
-                Task::DispatchDueEvents => self.dispatch_due_events(),
+    fn get_next_task(&mut self) -> Option<Task> {
+        let now = precise_time_ns();
+        if let Some(event) = self.scheduled_events.peek() {
+            if event.for_time < now {
+                return Some(Task::DispatchDueEvents);
             }
+
+            let timeout = now - event.for_time;
```

Comment:
This is wrong, and should be the inverse, I'll fix it when I'm in front of a computer